fog/sample-paykit/src/cached_tx_data/memo_handler.rs

```diff
@@ -120,6 +120,35 @@ impl MemoHandler {
                 // TODO: Add Gift Code Validation
                 Ok(Some(memo_type))
             }
+            MemoType::AuthenticatedSenderWithPaymentIntentId(memo) => {
+                if let Some(addr) = self.contacts.get(&memo.sender_address_hash()) {
+                    if bool::from(memo.validate(
+                        addr,
+                        &account_key.default_subaddress_view_private(),
+                        &tx_out.public_key,
+                    )) {
+                        Ok(Some(memo_type))
+                    } else {
+                        Err(MemoHandlerError::FailedHmacValidation)
+                    }
```

Comment:
Optional/Equivalent: Use `map`
```suggestion
                    memo.validate(
                        addr,
                        &account_key.default_subaddress_view_private(),
                        &tx_out.public_key,
                    ).map(|_| Some(memo_type))
                    .map_err(|_| MemoHandlerError::FailedHmacValidation)
```
(probably needs formatting)

Reply:
Sorry, I thought `validate()` returned a `Result`